File: Template.scala
import java.io._
import java.util._

object Template {

  def main(args: Array[String]) {
    val name = "A-small"
    val path = ""

    Locale.setDefault(Locale.US)

    val sc = new Scanner(new File(path + name + ".in"))
    val pw = new PrintWriter(path + name + ".out")

//    val sc = new Scanner(System.in)
//    val pw = new PrintWriter(System.out)

    val testCases = sc.nextInt

    for (testCase <- 1 to testCases) {
      val cmd = sc.next
      val n : Int = sc.nextInt
      val a = Seq.fill(n)(sc.nextDouble)

      val res = cmd match {
        case "median" => a.sorted.apply(n / 2)
        case "mean" => a.sum / n
      }

      pw.println(s"Case #${testCase}: " + "%.10f".format(res))
      pw.flush()
    }

    pw.close()
    sc.close()
  }
}

File: Template.java
import java.io.*;
import java.util.*;

public class Template {

	public static void main(String[] args) throws Exception {
		String name = "A-small";
		String path = "";

		Locale.setDefault(Locale.US);

		Scanner sc = new Scanner(new File(path + name + ".in"));
		PrintWriter pw = new PrintWriter(path + name + ".out");
//		Scanner sc = new Scanner(System.in);
//		PrintWriter pw = new PrintWriter(System.out);

		int testCases = sc.nextInt();
		for (int testCase = 1; testCase <= testCases; testCase++) {
			String cmd = sc.next();
			int n = sc.nextInt();
			double[] a = new double[n];
			for (int i = 0; i < n; i++) {
				a[i] = sc.nextDouble();
			}

			final double res;
			switch (cmd) {
				case "median":
					Arrays.sort(a);
					res = a[n / 2];
					break;
				case "mean":
					double sum = 0;
					for (double v : a) {
						sum += v;
					}
					res = sum / n;
					break;
				default:
					throw new RuntimeException();
			}
			
			pw.printf("Case #" + testCase + ": %.10f\n", res);
			pw.flush();
		}

		pw.close();
		sc.close();
	}
}
